modules/search/src/main/java/com/sygic/maps/module/search/SearchFragment.kt

```diff
@@ -144,11 +152,46 @@ class SearchFragment : Fragment(), SdkInitializationManager.Callback {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        fragmentViewModel = ViewModelProviders.of(this, viewModelFactory)[SearchFragmentViewModel::class.java]
-        searchToolbarViewModel = ViewModelProviders.of(this, viewModelFactory.with(searchToolbarInitComponent))[SearchToolbarViewModel::class.java]
+        fragmentViewModel = ViewModelProviders.of(
+            this,
+            viewModelFactory.with(searchFragmentInitComponent)
+        )[SearchFragmentViewModel::class.java].apply {
+            this.hideKeyboardObservable.observe(
+                this@SearchFragment,
+                Observer<Any> { hideKeyboard() })
+            this.popBackStackObservable.observe(
+                this@SearchFragment,
+                Observer<Any> { fragmentManager?.popBackStack() })
+        }
+        searchToolbarViewModel = ViewModelProviders.of(
+            this,
+            viewModelFactory.with(searchToolbarInitComponent)
+        )[SearchToolbarViewModel::class.java].apply {
+            this.onActionSearchClickObservable.observe(
+                this@SearchFragment,
+                Observer<Any> { fragmentViewModel.onActionSearchClick() })
+        }
+        searchResultListViewModel = ViewModelProviders.of(
+            this,
+            viewModelFactory
+        )[SearchResultListViewModel::class.java].apply {
+            this.removeFocusAndHideKeyboardObservable.observe(
```

Comment:
tu by som zase zvolil nejaky generickejsi nazov, lebo ak pouzijes result list (view model) a pouzijes ho bez toolbaru, tak meno `removeFocusAndHideKeyboardObservable` nebude davat zmysel. malo by to skor opisovat udalost co sa stala a teda nieco na styl `onListScrolled` alebo `onListInteraction`